core/server/data/migrations/versions/4.35/2022-02-02-13-10-transform-specific-tiers-default-content-visibility.js

```diff
@@ -0,0 +1,147 @@
+const logging = require('@tryghost/logging');
+
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Checking default_content_visibility for specific tiers');
+
+        const settings = await knex('settings')
+            .select()
+            .whereIn('key', ['default_content_visibility', 'default_content_visibility_tiers']);
+        const contentVisibilitySetting = settings.find(d => d.key === 'default_content_visibility');
+        const visibilityTiersSetting = settings.find(d => d.key === 'default_content_visibility_tiers');
+        if (!contentVisibilitySetting || !contentVisibilitySetting.value) {
```

Comment:
If `default_content_visibility` === `''` (empty string), we'll stop here, but we actually want to reset the value to `paid` below right?

Reply:
🤔 umm so this is a tricky one, so i had done this intentionally thinking that a site must have really messed up to have an empty `default_content_visibility` as there is still a valid path to invalid filter but none to this. My thinking here was if we are not creating the setting here if its not found, we shouldn't also update the tiers value if a site managed to set it empty.

I'll remove the `|| !contentVisibilitySetting.value` from this condition so empty string is treated the same as invalid filter as on second thought that does seem a bit more consistent 👍